diff -ru src.orig/port/linux/include/port_after.h src/port/linux/include/port_after.h
--- src.orig/port/linux/include/port_after.h	2002-04-28 01:12:52.000000000 +0000
+++ src/port/linux/include/port_after.h	2004-11-24 18:38:18.000000000 +0000
@@ -13,9 +13,9 @@
 #define HAVE_CHROOT
 #define CAN_CHANGE_ID
 
-#if __GLIBC__ >= 2
+#if __GLIBC__ >= 2 || defined(__UCLIBC__)
 #  define dprintf bind_dprintf
-#  if __GLIBC_MINOR__ < 1
+#  if __GLIBC_MINOR__ < 1 || defined(__UCLIBC__)
 #    define NEED_PSELECT
 #    define pselect bind_pselect
 #  endif
